main/acle.md

```diff
@@ -4847,16 +4874,34 @@ interfaces:
 
 *   a “shared-ZA” interface
 
-If a C or C++ function F forms part of the object code's ABI, that
-object code function has a shared-ZA interface if and only if at least
-one of the following is true:
+<span id="agnostic-za"></span>
+
+*   an "agnostic-ZA" interface
 
-*   F shares ZA with its caller
+If a C or C++ function F forms part of the object code's ABI:
 
-*   F shares ZT0 with its caller
+* the object code function has a shared-ZA interface if and only if at least
+  one of the following is true:
+
+  * F shares ZA with its caller
+
+  * F shares ZT0 with its caller
+
+* the object code function has an agnostic-ZA interface if and only if F's type
+  has an `__arm_agnostic("sme_za_state")` attribute.
 
 All other functions have a private-ZA interface.
 
+If F implements an agnostic-ZA interface and ZA is active, then a call to F must
```

Comment:
I still think we should drop these two paragraphs.  IMO the purpose of this section is to describe how ACLE features map to the ABI, and it's the ABI's job to describe the actual binary interface.

I was hoping the new “The use of this attribute does not imply that SME is available.” statement would replace the second paragraph, since IMO it's more precise.  E.g. it's valid to do:

```
  void f() __arm_agnostic("sme_state") __arm_streaming {}
```

and in that case, the implementation _can_ assume that PSTATE.ZA is available.

In other words, the point is that `__arm_agnostic` does not _by itself_ guarantee that SME is available, but the compiler might still know that SME is available from other sources, e.g. from `+sme` or from other attributes.  `__arm_agnostic` doesn't override information from other sources.